controllers/policies/policyserver_controller.go

```diff
@@ -152,71 +192,43 @@ func (r *PolicyServerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	err = ctrl.NewControllerManagedBy(mgr).
 		For(&policiesv1alpha2.PolicyServer{}).
 		WithEventFilter(predicate.GenerationChangedPredicate{}).
-		Watches(&source.Kind{Type: &policiesv1alpha2.ClusterAdmissionPolicy{}}, handler.EnqueueRequestsFromMapFunc(func(object client.Object) []reconcile.Request {
-			// The watch will trigger twice per object change; once with the old
-			// object, and once the new object. We need to be mindful when doing
-			// Updates since they will invalidate the newever versions of the
-			// object.
-			policy, ok := object.(*policiesv1alpha2.ClusterAdmissionPolicy)
-			if !ok {
-				r.Log.Error(err, "object is not type of ClusterAdmissionPolicy: %#v", policy)
-				return []ctrl.Request{}
-			}
-
-			if policy.Status.PolicyStatus == "" {
-				policy.Status.PolicyStatus = policiesv1alpha2.ClusterAdmissionPolicyStatusUnscheduled
-				err := r.Reconciler.UpdateAdmissionPolicyStatus(context.Background(), policy)
-				if err != nil {
-					r.Log.Error(err, "cannot update status of policy "+policy.Name)
-					return []ctrl.Request{}
+		Watches(&source.Kind{Type: &policiesv1alpha2.ClusterAdmissionPolicy{}}, handler.Funcs{
+			CreateFunc: func(e event.CreateEvent, queue workqueue.RateLimitingInterface) {
+				queue.Add(watchClusterAdmissionPolicy(r, e.Object))
+			},
+			UpdateFunc: func(e event.UpdateEvent, queue workqueue.RateLimitingInterface) {
+				newPolicy, ok := e.ObjectNew.(*policiesv1alpha2.ClusterAdmissionPolicy)
+				if !ok {
+					r.Log.Error(err, "ObjectNew is not type of ClusterAdmissionPolicy: %#v", newPolicy)
+					queue.Add(ctrl.Request{})
+					return
 				}
```

Comment:
I find a little strange checking for the type of the object in some places before adding to the queue and not on some others, is this in purpose? Should we keep this behavior or remove this behavior and behave in the same way in all cases?

E.g. `CreateFunc` vs `UpdateFunc` vs `DeleteFunc` vs `GenericFunc` on this `handler.Funcs` instance.

Reply:
No, this is not in purpose. I missed this and I'll push this changes soon.